gson/src/test/java/com/google/gson/functional/Java17RecordTest.java

```diff
@@ -43,28 +42,42 @@ public void testMultipleNamesDeserializedCorrectly() {
     assertEquals("v1", gson.fromJson("{'name':'v1'}", MyRecord.class).a);
 
     // Both name1 and name2 gets deserialized to b
-    assertEquals("v11", gson.fromJson("{'name1':'v11'}", MyRecord.class).b);
-    assertEquals("v2", gson.fromJson("{'name2':'v2'}", MyRecord.class).b);
-    assertEquals("v3", gson.fromJson("{'name3':'v3'}", MyRecord.class).b);
+    assertEquals("v11", gson.fromJson("{'name': 'v1', 'name1':'v11'}", MyRecord.class).b);
+    assertEquals("v2", gson.fromJson("{'name': 'v1', 'name2':'v2'}", MyRecord.class).b);
+    assertEquals("v3", gson.fromJson("{'name': 'v1', 'name3':'v3'}", MyRecord.class).b);
   }
 
   @Test
   public void testMultipleNamesInTheSameString() {
     // The last value takes precedence
-    assertEquals("v3", gson.fromJson("{'name1':'v1','name2':'v2','name3':'v3'}", MyRecord.class).b);
+    assertEquals("v3",
+        gson.fromJson("{'name': 'foo', 'name1':'v1','name2':'v2','name3':'v3'}", MyRecord.class).b);
   }
 
   @Test
   public void testConstructorRuns() {
-    assertThrows(NullPointerException.class,
-        () -> gson.fromJson("{'name1': null, 'name2': null", MyRecord.class));
+    Throwable e = assertThrows(RuntimeException.class,
+        () -> gson.fromJson("{'name1': null, 'name2': null}", MyRecord.class));
+    while (e != null && !(e instanceof NullPointerException)) {
+      e = e.getCause();
+    }
+    assertNotNull(e);
```

Comment:
Rather than using an exception to signal that the constructor was called it might be a bit cleaner to just have the constructor modify the provided value, e.g.:
```java
public record MyRecord(String a) {
  public MyRecord {
    a = "constructor-" + a;
  }
}
```

Reply:
Yes, the exception stuff was clunky. I think I was actually using this test to check how to rewrite `@AutoValue` classes as records, which is hardly the right place for it. Done.